torch/ao/quantization/pt2e/_numeric_debugger.py

```diff
@@ -163,26 +201,16 @@ class QuantizationComparisonResult:
 
     @property
     def mse_loss(self) -> torch.Tensor:
-        return F.mse_loss(
-            self.actual.to(dtype=torch.float32), self.ref.to(dtype=torch.float32)
-        )
+        return self.loss(F.mse_loss)
 
     @property
     def sqnr(self) -> torch.Tensor:
-        return compute_sqnr(
-            self.actual.to(dtype=torch.float32), self.ref.to(dtype=torch.float32)
-        )
+        return self.loss(compute_sqnr)
 
     def loss(
         self, loss_function: Callable[[torch.Tensor, torch.Tensor], torch.Tensor]
     ) -> torch.Tensor:
-        if self.actual.shape != self.ref.shape:
-            raise ValueError(
-                f"Cannot compare tensors with different shapes: {self.actual.shape} vs {self.ref.shape}"
-            )
-        return loss_function(
-            self.actual.to(dtype=torch.float32), self.ref.to(dtype=torch.float32)
-        )
+        return _loss_fn(loss_function, self.actual, self.ref)
```

Comment:
You might want to leave in the `to(torch.float32)`, as most loss functions require floats.

Otherwise `self.loss(F.mse_loss)` fails for integer tensors